numalogic/models/autoencoder/pipeline.py

```diff
@@ -147,18 +149,18 @@ def fit(self, X: NDArray[float], y=None, log_freq: int = 5) -> "AutoencoderPipel
         dataset = self._model.construct_dataset(X, self.seq_len)
         loader = DataLoader(dataset, batch_size=self.batch_size, shuffle=False)
         self._model.train()
-
-        loss = torch.Tensor([0.0])
+        losses = []
         for epoch in range(1, self.num_epochs + 1):
             for x_batch in loader:
                 self.optimizer.zero_grad()
                 _, decoded = self._model(x_batch)
                 loss = self.criterion(decoded, x_batch)
                 loss.backward()
                 self.optimizer.step()
-
+                losses.append(loss.item())
             if epoch % log_freq == 0:
-                _LOGGER.info(f"epoch : {epoch}, loss_mean : {loss.item():.7f}")
+                _LOGGER.info(f"epoch : {epoch}, loss_mean : {np.array(losses).mean():.7f}")
```

Comment:
Nitpick. 
We can just directly do `np.mean(losses)`

Reply:
updated!